packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -307,6 +323,16 @@ export class NodeProxyTarget extends ProxyTarget<Anchor> {
         return this.cursor.value;
     }
 
+    set value(value: Value) {
+        assert(
+            isPrimitive(this.type) && this.type.value !== ValueSchema.Serializable,
+            "Cannot set a value of a non-primitive field",
+        );
```

Comment:
This isn't the right validation.
We should test if the value is permitted by the ValueSchema (ex: only allow objects if ValueSchema.Serializable, only allow numbers if ValueSchema.Number etc.)

I don't think we currently have a function to do that, but you could add one and call it from here.

Reply:
Well, ideally it should be validated even before by a static type, but we can introduce also a schema validation for primitives here. As for `ValueSchema.Serializable`, I'd set a more specific "not supported" failure, as it seems that currently it is not supported by the changeset.

Reply:
Fixed, although still forbidden for non-primitive `object`, as it will require to completely rework the way we currently handle serializable.